src/Components/3DV/SceneView.tsx

```diff
@@ -1680,6 +1683,132 @@ function SceneView(props: ISceneViewProps, ref) {
         };
     }, [outlinedMeshitems, meshMap.current]);
 
+    // SETUP LOGIC FOR HANDLING TRANSFORMING MESHES
+    useEffect(() => {
+        debugLog(
+            'debug',
+            'transform meshes based on transformedMeshItems prop' +
+                (scene ? ' with scene' : ' no scene')
+        );
+
+        if (scene && transformedElementItems && !isLoading) {
+            if (debugLogging) {
+                console.time('transforming meshes');
+            }
+            try {
+                // if there is a parent mesh in previouslyTransformedElements BUT NOT in transformedElementItems
+                // (meaning the element had been previously transformed but the transform is now turned off)
+                // reset the element to its original state (which had been preserved in previouslyTransformedElements)
+                // and remove element from previouslyTransformedElements
+
+                // grab all parentMeshIds from the new transformedElementsItems
+                const tEIParentMeshIds = transformedElementItems.map(
+                    (tEI) => tEI.parentMeshId
+                );
+                // iterate through all previously transformed elements
+                previouslyTransformedElements.current.forEach(
+                    (previouslyTransformedElement) => {
+                        // get mesh id for the previously transformed element
+                        const prevTransParentMeshId =
+                            previouslyTransformedElement.meshId;
+                        // if the parentMeshIds to be transformed DOES NOT include the previouslyTransformedParentMeshId,
+                        // then that means the transformation no longer applies and the element should be reset
+                        if (!tEIParentMeshIds.includes(prevTransParentMeshId)) {
+                            const prevTransParentMesh: BABYLON.Mesh =
+                                meshMap.current?.[prevTransParentMeshId];
+                            // setting rotation back to original
+                            if (prevTransParentMesh) {
```

Comment:
could we pull some of these things out to the `SceneView.Utils` file? Like take in the 2 objects but do all the mapping of properties in the uility to simplify this file?

Reply:
done! (also used them to refactor gizmo functions)